lib/r7_insight/host/connection.rb

```diff
@@ -188,9 +188,17 @@ def open_connection
             ssl_context = OpenSSL::SSL::SSLContext.new
             ssl_context.cert_store = cert_store
 
-            ssl_context.min_version = OpenSSL::SSL::TLS1_1_VERSION
-            # For older versions of openssl (prior to 1.1.1) missing support for TLSv1.3
-            ssl_context.max_version = defined?(OpenSSL::SSL::TLS1_3_VERSION) ? OpenSSL::SSL::TLS1_3_VERSION : OpenSSL::SSL::TLS1_2_VERSION
+            if defined?(ssl_context.min_version)
```

Comment:
Great spot but small thing, the if doesn't trigger with new OpenSSL, I believe the correct check is `OpenSSL::SSL::SSLContext.method_defined? :min_version=`
With this test script
```ruby
require 'openssl'

ssl_context = OpenSSL::SSL::SSLContext.new
if OpenSSL::SSL::SSLContext.method_defined? :min_version=
  print 'min'
  ssl_context.min_version = OpenSSL::SSL::TLS1_1_VERSION
end
if OpenSSL::SSL::SSLContext.method_defined? :max_version=
  print 'max'
  # For older versions of openssl (prior to 1.1.1) missing support for TLSv1.3
  ssl_context.max_version = if defined?(OpenSSL::SSL::TLS1_3_VERSION)
                              OpenSSL::SSL::TLS1_3_VERSION
                            else
                              OpenSSL::SSL::TLS1_2_VERSION
                            end
end
```

I ran the following:
```bash
~/.../r7insight_ruby (ensure-ssl-version) $
~/.../r7insight_ruby (ensure-ssl-version) $ rvm use 2.5.0
Using /Users/sbialkowski/.rvm/gems/ruby-2.5.0
~/.../r7insight_ruby (ensure-ssl-version) $ ruby test.rb
minmax~/.../r7insight_ruby (ensure-ssl-version) $
~/.../r7insight_ruby (ensure-ssl-version) $
~/.../r7insight_ruby (ensure-ssl-version) $ rvm use 2.4.1
Using /Users/sbialkowski/.rvm/gems/ruby-2.4.1
~/.../r7insight_ruby (ensure-ssl-version) $
~/.../r7insight_ruby (ensure-ssl-version) $
~/.../r7insight_ruby (ensure-ssl-version) $ ruby test.rb
~/.../r7insight_ruby (ensure-ssl-version) $ # Correctly doesn't set min/max
~/.../r7insight_ruby (ensure-ssl-version) $
```

No errors on both 2.5.0 and 2.4.1

Reply:
Sounds good!